aries/aries_vcx/src/utils/encryption_envelope.rs

```diff
@@ -88,17 +97,24 @@ impl EncryptionEnvelope {
     pub async fn create_from_keys(
         wallet: &impl BaseWallet,
         data: &[u8],
-        sender_vk: Option<&str>,
-        // TODO - why not have encryption envelope take typed [Key]s, and enforce they are
-        // KeyType::Ed25519
-        recipient_key: String,
-        routing_keys: Vec<String>,
+        sender_vk: Option<Key>,
+        recipient_key: Key,
+        routing_keys: Vec<Key>,
```

Comment:
nit; can all or any of these be passed by reference instead of taking ownership? e.g. `Option<&Key>`, `&Key`, `&[Key]`

Reply:
general comment from around the codebase, IMO it's best to not make consumers give up ownership unless we need it

Reply:
I improved this somewhat--ended up being challenging to make many improvements here as the underlying code getting called for packing and unpacking is the did_wallet trait and its implementations, and I didn't really want to go down that rabbit hole tonight (especially as it may end up going all the way down to Askar itself). Also tried to use [u8] where possible, but same story as above.